software/Test_Robot_Software/find_a_ball.py

```diff
@@ -0,0 +1,100 @@
+import image_processor
+import camera
+import motion
+import cv2
+import time
+import api
+
+def main_loop():
+    debug = True
+    
+    cam = camera.RealsenseCamera(exposure = 100)    
+    processor = image_processor.ImageProcessor(cam, debug=debug)
+    processor.start()
+
+    start = time.time() 
+    fps = 0
+    frame = 0
+    frame_cnt = 0
+    
+    scaled_pos_x = 0.0
+    scaled_pos_y = 0.0
+
+    try:
+        while True:
+            
+            # has argument aligned_depth that enables depth frame to color frame alignment. Costs performance
+            processedData = processor.process_frame(aligned_depth=False)
+            largest = max(processedData.balls, key = lambda ball: ball.size, default=None)
+            if(largest):
+                cv2.circle(processedData.debug_frame,(largest.x, largest.y), 20, (255, 0, 255), -1)
+                                
+                scaled_pos_x = (largest.x - processedData.debug_frame.shape[1]/2) / (processedData.debug_frame.shape[1]/2) 
```

Comment:
You can access frame sizes from the camera object as well.

Reply:
fetched the frame width and height from camera object